### ISSUE TITLE
[Ecosystem] TensorLy-Torch

### ISSUE THREAD

**Issue by JeanKossaifi** [pytorch-fdn/ecosystem#41]
### Contact emails

jean.kossaifi@gmail.com

### Project summary

Deep Tensor Learning with TensorLy and PyTorch

### Project description

TensorLy-Torch is a Python library for deep tensor networks that builds on top of TensorLy and PyTorch. It allows to easily leverage tensor methods in a deep learning setting and comes with all batteries included.

Tensor methods generalize matrix algebraic operations to higher-orders. Deep neural networks typically map between higher-order tensors. In fact, it is the ability of deep convolutional neural networks to preserve and leverage local structure that, along with large datasets and efficient hardware, made the current levels of performance possible. Tensor methods allow to further leverage and preserve that structure, for individual layers or whole networks.

In TensorLy-Torch, we provide convenient layers that do all the heavy lifting for you and provide the benefits tensor based layers wrapped in a nice, well documented and tested API.

It provides:
* **Tensor Factorizations**: decomposing, manipulating and initializing tensor decompositions can be tricky. We take care of it all, in a convenient, unified API.
* **Leverage structure in your data**: with tensor layers, you can easily leverage the structure in your data, through Tensor Regression Layers, Factorized Convolutions, etc
* **Built-in tensor layers:** all you have to do is import tensorly torch and include the layers we provide directly within your PyTorch models!
* **Tensor hooks**: you can easily augment your architectures with our built-in Tensor Hooks. Robustify your network with Tensor Dropout and automatically select the rank end-to-end with L1 Regularization
* **All the methods available**: we are always adding more methods to make it easy to compare between the performance of various deep tensor based methods

### Are there any other projects in the PyTorch Ecosystem similar to yours?  If, yes, what are they?

N\A

### Project repo URL

https://github.com/tensorly/torch

### Project license

BSD-3 Clause

### GitHub handles of the project maintainer(s)

JeanKossaifi, animakumar

### Website URL

https://github.com/tensorly/torch

### Documentation

API: https://tensorly.org/torch/dev/modules/api.html
User-guide: https://tensorly.org/torch/dev/user_guide/index.html

### How do you build and test the project today (continuous integration)?  Please describe.

CI using Github actions and pytest, thorough unit-tests and coverage.

### Version of PyTorch

Supporting most versions, focusing on the cutting-edge

### Components of PyTorch

Most components, from base Tensor classes (which we support via __torch_function__), to functional and nn modules. We provide PyTorch-like factorized tensors that can be efficiently manipulated as well as nn and functional modules leveraging tensor methods that can be directly plugged in. We also support efficient initialization as well as custom hooks for things such as tensor dropout and rank regularization. 

### How long do you expect to maintain the project?

We plan to continue maintaining the project to support the adoption of tensor methods in PyTorch.

**Comment by JeanKossaifi** [comment]
Thanks @lantiga! Yes, it is part of the TensorLy ecosystem so the algorithms themselves are updated directly in TensorLy, TensorLy-Torch provides an interface to use these transparently as deep learning layers in PyTorch. We are planning to continue supporting it - we actually have a big update in the works!

**Comment by michellearoth** [comment]
@JeanKossaifi closing this for now but please let us know if you would like to reopen.

**Comment by JeanKossaifi** [comment]
Thanks @michellearoth  - will do as soon as we have updates to share!
